src/Analysis/Engine/Impl/Interpreter/Ast/AstTypeAnnotationConverter.cs

```diff
@@ -196,6 +192,11 @@ class AstTypeAnnotationConverter : TypeAnnotationConverter<IPythonType> {
             return res;
         }
 
+        private IPythonType MakeGenericClassType(IPythonType typeArg)
+            => typeArg.IsBuiltin
```

Comment:
`UnionType` always return `IsBuiltin => false`. Is that ok?

Reply:
I changed implementation, will update soon